PhysicsTools/NanoAOD/python/electrons_cff.py

```diff
@@ -250,53 +258,22 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
         jetForLepJetVar = cms.InputTag("ptRatioRelForEle:jetForLepJetVar") # warning: Ptr is null if no match is found
     ),
 )
-for modifier in run2_egamma_2017,run2_egamma_2018, run2_miniAOD_80XLegacy,run2_nanoAOD_94XMiniAODv1,run2_nanoAOD_94XMiniAODv2,run2_nanoAOD_94X2016,run2_nanoAOD_102Xv1,run2_nanoAOD_106Xv1:
+for modifier in run2_egamma_2016,run2_egamma_2017,run2_egamma_2018, run2_miniAOD_80XLegacy,run2_nanoAOD_94XMiniAODv1,run2_nanoAOD_94XMiniAODv2,run2_nanoAOD_94X2016,run2_nanoAOD_102Xv1,run2_nanoAOD_106Xv1:
     modifier.toModify(slimmedElectronsWithUserData, src = "slimmedElectronsUpdated")
-
-
-run2_egamma_2017.toModify(slimmedElectronsWithUserData.userFloats,
-    ecalTrkEnergyErrPostCorrNew = cms.InputTag("calibratedPatElectronsUL17","ecalTrkEnergyErrPostCorr"),
-    ecalTrkEnergyPreCorrNew     = cms.InputTag("calibratedPatElectronsUL17","ecalTrkEnergyPreCorr"),
-    ecalTrkEnergyPostCorrNew    = cms.InputTag("calibratedPatElectronsUL17","ecalTrkEnergyPostCorr"),
-)
-
-run2_egamma_2018.toModify(slimmedElectronsWithUserData.userFloats,
-    ecalTrkEnergyErrPostCorrNew = cms.InputTag("calibratedPatElectronsUL18","ecalTrkEnergyErrPostCorr"),
-    ecalTrkEnergyPreCorrNew     = cms.InputTag("calibratedPatElectronsUL18","ecalTrkEnergyPreCorr"),
-    ecalTrkEnergyPostCorrNew    = cms.InputTag("calibratedPatElectronsUL18","ecalTrkEnergyPostCorr"),
-)
+    modifier.toModify(slimmedElectronsWithUserData.userFloats,
+                      ecalTrkEnergyErrPostCorrNew = cms.InputTag("calibratedPatElectronsNano","ecalTrkEnergyErrPostCorr"),
+                      ecalTrkEnergyPreCorrNew     = cms.InputTag("calibratedPatElectronsNano","ecalTrkEnergyPreCorr"),
+                      ecalTrkEnergyPostCorrNew    = cms.InputTag("calibratedPatElectronsNano","ecalTrkEnergyPostCorr"),
+                      energyScaleUp               = cms.InputTag("calibratedPatElectronsNano","energyScaleUp"),
+                      energyScaleDown             = cms.InputTag("calibratedPatElectronsNano","energyScaleDown"),
+                      energySigmaUp               = cms.InputTag("calibratedPatElectronsNano","energySigmaUp"),
+                      energySigmaDown             = cms.InputTag("calibratedPatElectronsNano","energySigmaDown"),
+    )
```

Comment:
These .userFloat do not exist for the run2_nanoAOD_94X2016, and are responsible for the failure.

In fact for the 80XLegacy you redefine these  .userFloat below and you should restore the old setting for the run2_nanoAOD_94X2016 as well.
For the run2_nanoAOD_94X2016 you correctly already restore the .userIntFromBools and .userInt.

Same happened for the photon.




Remove from the general list in line 261 the run2_miniAOD_80XLegacy and run2_nanoAOD_94X2016

Reply:
Thanks Maria, for run2_miniAOD_80XLegacy and run2_nanoAOD_94X2016 I think we have to keep since the scale and smearing corrections existed. We just need to run the calibratedElectron path for run2_nanoAOD_94X2016 as well - will update.